firebase-perf/src/main/java/com/google/firebase/perf/metrics/validator/FirebasePerfTraceValidator.java

```diff
@@ -180,9 +180,10 @@ private boolean isValidCounterId(@Nullable String counterId) {
 
   private boolean hasValidAttributes(Map<String, String> customAttributes) {
     for (Map.Entry<String, String> entry : customAttributes.entrySet()) {
-      String err = PerfMetricValidator.validateAttribute(entry);
-      if (err != null) {
-        logger.warn(err);
+      try {
+        PerfMetricValidator.validateAttribute(entry);
+      } catch (IllegalArgumentException exception) {
+        logger.warn(exception.getLocalizedMessage());
```

Comment:
nit: (preferably in a separate cl) we should have a separate logger.warn that accepts a throwable. This way the stacktrace is still logged.

https://developer.android.com/reference/android/util/Log#w(java.lang.String,%20java.lang.Throwable)

Reply:
Acknowledged! Good idea and will follow up on this separately.